crypto/provider_core.c

```diff
@@ -1451,7 +1450,16 @@ int ossl_provider_doall_activated(OSSL_LIB_CTX *ctx,
     for (curr++; curr < max; curr++) {
         OSSL_PROVIDER *prov = sk_OSSL_PROVIDER_value(provs, curr);
 
-        provider_deactivate(prov, 0, 1);
+        CRYPTO_DOWN_REF(&prov->activatecnt, &ref, prov->refcnt_lock);
```

Comment:
Hmm... we do not check the return value here?

(Fallible locking & atomic fallbacks are indeed a pain...)

Reply:
Fixed.